config/v1/types_cluster_version.go

```diff
@@ -406,8 +420,19 @@ type ComponentOverride struct {
 type URL string
 
 // Update represents an administrator update request.
+// +kubebuilder:validation:XValidation:rule="has(self.architecture) && has(self.version) ? (size(self.architecture) == 0 || size(self.version) == 0) : true",message="cannot set both Architecture and Version"
```

Comment:
this seems to be missing a test. Have a look at how to create a testing yaml in the example api group.

Reply:
added test